File: 06_Inheritance/Super/Main.java
class Person {
    private String name;

    public Person(String name) {
        this.name = name;
    }

    public void display() {
        System.out.println("Name: " + name);
    }
}

class Employee extends Person {
    private String company;

    public Employee(String name, String company) {
        super(name);
        this.company = company;
    }

    @Override
    public void display() {
        super.display(); 
        System.out.println("Company: " + company);
    }
}

public class Main {
    public static void main(String[] args) {
        Employee employee = new Employee("Alice", "TechCorp");
        employee.display();
    }
}

File: 06_Inheritance/Super/Main.cpp
#include <iostream>
#include <string>
using namespace std;

class Person {
private:
    string name;

public:
    Person(string name) : name(name) {}

    void display() {
        cout << "Name: " << name << endl;
    }
};

class Employee : public Person {
private:
    string company;

public:
    Employee(string name, string company) : Person(name) {
        this->company = company;
    }

    void display() {
        Person::display(); 
        cout << "Company: " << company << endl;
    }
};

int main() {
    Employee employee("Alice", "TechCorp");
    employee.display();
    return 0;
}
